compiler/rustc_error_codes/src/error_codes/E0755.md

```diff
@@ -0,0 +1,27 @@
+The `ffi_pure` attribute was used on a non-foreign function.
+
+Erroneous code example:
+
+```compile_fail,E0755
+#![feature(ffi_pure)]
+
+#[ffi_pure] // error!
+pub fn foo() {}
+# fn main() {}
+```
+
+The `ffi_pure` attribute can only be used on foreign functions which do not have
+effects except for their return value. You can find more information about it in
+the [unstable Rust Book].
```

Comment:
```suggestion
side effects. You can find more information about it in the [unstable Rust Book].
```
Returning value is probably not counted as side effects.

Reply:
I quoted the book.

Reply:
Yes, I took from the book too.

Reply:
Then perfect. :)